pkg/plugin/framework/server_mux.go

```diff
@@ -43,9 +45,12 @@ func newServerMux(logger logrus.FieldLogger) *serverMux {
 	}
 }
 
-// register registers the initializer for name.
+// register registers the initializer for the given name. Proper format
+// for the name is <namespace>/<name> and there can be no duplicates.
 func (m *serverMux) register(name string, f HandlerInitializer) {
-	// TODO(ncdc): return an error on duplicate registrations for the same name.
+	if !validPluginName(name, m.names()) {
```

Comment:
I think it makes sense to have this check here, on the server side; additionally, I think we need a similar check in `clientmgmt/registry.go`, on the client side -- probably in this function https://github.com/heptio/velero/blob/master/pkg/plugin/clientmgmt/registry.go#L199